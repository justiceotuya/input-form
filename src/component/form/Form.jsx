import React, { useState, useEffect } from 'react'
import { FullNameInput } from './fullNameInput'
import { StyledInputEmail } from './style'
export const Form = () => {
    //fullName state with rules that the state must pass
    const [fullName, setFullName] = useState({
        FullNameValue: '',
        rules: [
            {
                id: 1,
                rule: "Name must not be less that 2 characters",
                isPassed: false
            },
            {
                id: 2,
                rule: "must include a space",
                isPassed: false
            },
            {
                id: 3,
                rule: "Full Name must include second name",
                isPassed: false
            }
        ],
        showFullNameValidationError: false
    })
    //show or hide validation error for fullname
    const [showFullNameValidationError, setShowFullNameValidationError] = useState(false)
    const [email, setEmail] = useState({
        value: '',
        rules: {
            rule: 'this must an email',
            isPassed: false
        }

    })
    const [showEmailValidationError, setShowEmailValidationError] = useState(false)

    useEffect(() => {
        // handles validation error for fullname field
        const handleFullNameValidationError = () => {
            const { FullNameValue } = fullName
            //get a copy of the fullname object
            let newState = { ...fullName }


            // makes sure that the fullname is more than two character
            if (FullNameValue.trim().length >= 2) {
                newState.rules[0].isPassed = true
                setFullName({ ...newState })
            } else {
                newState.rules[0].isPassed = false
                setFullName({ ...newState })
            }


            //check if the value of fullname with a length greater than zero and has space, then switch the isPassed flag. makes sure the word Name includes a space
            if (FullNameValue.length > 0 && FullNameValue.includes(' ')) {
                newState.rules[1].isPassed = true
                setFullName({ ...newState })
            } else {
                newState.rules[1].isPassed = false
                setFullName({ ...newState })
            }

            // checks the word count and make sure the lenght is not less than two
            let wordCount = FullNameValue.split(' ').filter(word => word !== '').length
            if (wordCount >= 2) {
                newState.rules[2].isPassed = true
                setFullName({ ...newState })
            } else {
                newState.rules[2].isPassed = false
                setFullName({ ...newState })
            }

            // check if the fullname is validated
            let isNotValidated = fullName.rules.some(rule => rule.isPassed === false)
            console.log({ isNotValidated })
        }

        handleFullNameValidationError()
    }, [fullName.FullNameValue, fullName.isPassed])

    useEffect(() => {
        const handleEmailValidity = () => {

        }
        handleEmailValidity()
    }, [email])

    const handleFullName = (e) => {
        setFullName({
            ...fullName, FullNameValue: e.target.value
        })
    }

    const handleEmail = (e) => {
        const handleEmailValidity = () => {
            return e.target.checkValidity() && /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(e.target.value)
        }
        // deep clone the object
        let newEmail = JSON.parse(JSON.stringify(email));
        newEmail.value = e.target.value;
        newEmail.rules.isPassed = handleEmailValidity();
        setEmail(newEmail)

    }

    return (
        <form>
            <FullNameInput
                showFullNameValidationError={showFullNameValidationError}
                setShowFullNameValidationError={setShowFullNameValidationError}
                handleFullName={handleFullName}
                fullName={fullName}
            />
            <StyledInputEmail
                showEmailValidationError={showEmailValidationError}
                isPassed={email.rules.isPassed}
            >
                <input
                    type="email"
                    name="email"
                    id="email"
                    value={email.value}
                    onChange={handleEmail}
                    onBlur={() => setShowEmailValidationError(false)}
                    onFocus={() => setShowEmailValidationError(true)}
                    placeholder="enter email"
                />
                <label>Email</label>
                <p>This must be an email</p>

            </StyledInputEmail>

            <input type="submit" value="submit" />
        </form>
    )
}
