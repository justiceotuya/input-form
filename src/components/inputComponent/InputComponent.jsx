import React from 'react';
import PropTypes from 'prop-types';
import { StyledInputRule, StyledInputContainer } from './style';

/**
 * This components wraps around input component and is used to create other components in the form
 * @prop {boolean} ValidationError - validation error, is true if there is an error
 * @prop {object} inputValue - returns the value, rules and  isPassed flag of the in put
 * @prop {boolean} inputValue.isPassed - returns true is a rule is passed
 * @prop {string} inputValue.value - the value of the input field
 * @prop {array} inputValue.rules - an object that holds multiple rules that the input can pass
 * @prop {string} inputValue.rules.id -the id of a rule in an array of rules
 * @prop {boolean} inputValue.rules.isPassed - the isPassed flag of a rule. run checks against the input to know its status
 * @prop {string} inputValue.rules.value - the value of the input, the rule is run against this value
 * @prop {string} inputValue.rule - a single rule, if a component has just one rule
 * @prop {string} handleValueChange - this function handles the input value change
 * @prop {func} setShowValidationError - this helps in toggling showing the validation error in an input component
 * @prop {string} type - the type of the input eg 'password', 'text'
 * @prop {number} name - The name of the input field,  used in connecting the input to its label
 * @prop {string} placeholder - This helps the user understand what is suppose to go into an input field
 * @prop {string} label - the label of the input, this helps user know hat field the input is for
 */
export const InputComponent = ({
    ValidationError,
    inputValue,
    handleValueChange,
    setShowValidationError,
    type,
    name,
    placeholder,
    label,
}) => {
    const {
        rule, isPassed, value, rules,
    } = inputValue;
    const handleSetValidationError = () => {
        if (isPassed) {
            setShowValidationError(false);
        } else {
            setShowValidationError(true);
        }

        if (rules) {
            return rules.some(item => {
                item.isPassed === false
                    ? setShowValidationError(true)
                    : setShowValidationError(false);
            });
        }
    };

    return (
        <StyledInputContainer
            ValidationError={ValidationError}
            isPassed={rules ? null : isPassed}
            className={name}
        >
            <label htmlFor={name}>{label}</label>
            <input
                type={type}
                name={name}
                required
                id={name}
                value={value}
                onChange={handleValueChange}
                placeholder={placeholder}
                onBlur={handleSetValidationError}
                onFocus={handleSetValidationError}
            />

            {
                rules ? (
                    <div
                        className="rule"
                    >
                        {
                            rules.map(item => {
                                const { id, rule, isPassed } = item;
                                return (
                                    <StyledInputRule
                                        key={id}
                                        className={`rule-${id}`}
                                        isPassed={isPassed}
                                    >
                                        {rule}
                                    </StyledInputRule>
                                );
                            })
                        }
                    </div>
                ) : <p>{rule}</p>
            }
        </StyledInputContainer>

    );
};

InputComponent.propTypes = {
    ValidationError: PropTypes.bool.isRequired,
    handleValueChange: PropTypes.func.isRequired,
    inputValue: PropTypes.shape({
        isPassed: PropTypes.bool,
        value: PropTypes.string,
        rules: PropTypes.arrayOf(
            PropTypes.shape({
                id: PropTypes.number,
                isPassed: PropTypes.bool,
                value: PropTypes.string,
            })
        ),
    }).isRequired,
    label: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    placeholder: PropTypes.string.isRequired,
    setShowValidationError: PropTypes.func.isRequired,
    type: PropTypes.string.isRequired,
};

InputComponent.defaultProps = {
    value: '',
};
