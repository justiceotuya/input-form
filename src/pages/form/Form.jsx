import React, { useState, useEffect, Component } from 'react';
import { useHistory } from 'react-router-dom';
import { fullNameRules, phoneNumberRules, strings } from './constants';
import { StyledForm, StyledMainContainer, StyledButton } from './style';
import {
    handleEmailValidity,
    handlePasswordValidity,
    isAValidNumber
} from '../../utils';

import {
    PhoneNumberInput,
    EmailInput,
    PasswordInput,
    ConfirmPasswordInput,
    CardNumberInput,
    ExpiryDateInput,
    PinInput,
    FullNameInput
} from './components';

const {
    PASSWORD_RULE,
    EMAIL_RULE,
    CONFIRM_PASSWORD_RULE,
    CARD_NUMBER_RULE,
    EXPIRY_DATE_RULE,
    PIN_RULE,
} = strings;

export const Form = () => {
    const history = useHistory();
    const [fullName, setFullName] = useState({
        rules: fullNameRules,
        showFullNameValidationError: false,
        value: '',
    });

    const [email, setEmail] = useState({
        isPassed: false,
        rule: EMAIL_RULE,
        value: '',
    });

    const [password, setPassword] = useState({
        isPassed: false,
        rule: PASSWORD_RULE,
        value: '',
    });

    const [confirmPassword, setConfirmPassword] = useState({
        isPassed: false,
        rule: CONFIRM_PASSWORD_RULE,
        value: '',
    });

    const [phoneNumber, setPhoneNumber] = useState({
        rules: phoneNumberRules,
        value: '',
    });

    const [cardNumber, setCardNumber] = useState({
        isPassed: false,
        rule: CARD_NUMBER_RULE,
        value: '',
    });
    const [expiryDate, setExpiryDate] = useState({
        isPassed: false,
        rule: EXPIRY_DATE_RULE,
        value: '',
    });

    const [pin, setPin] = useState({
        isPassed: false,
        rule: PIN_RULE,
        value: '',
    });

    // show or hide validation error for fullname
    const [showFullNameValidationError,
        setShowFullNameValidationError] = useState(false);
    const [showEmailValidationError,
        setShowEmailValidationError] = useState(false);
    const [showPasswordValidationError,
        setShowPasswordValidationError] = useState(false);
    const [showConfirmPasswordValidationError,
        setShowConfirmPasswordValidationError] = useState(false);
    const [showPhoneNumberValidationError,
        setShowPhoneNumberValidationError] = useState(false);
    const [showCardNumberValidationError,
        setShowCardNumberValidationError] = useState(false);
    const [showExpiryValidationError,
        setShowExpiryValidationError] = useState(false);
    const [showPinValidationError,
        setShowPinValidationError] = useState(false);
    const [hasAllFieldPassedValidation,
        setHasAllFieldPassedValidation] = useState(false);

    useEffect(() => {
        // handles validation error for fullname field
        const handleFullNameValidationError = () => {
            const { value } = fullName;
            // get a copy of the fullname object
            const newState = { ...fullName };
            // makes sure that the fullname is more than two character
            if (value.trim().length >= 2) {
                newState.rules[0].isPassed = true;
                setFullName({ ...newState });
            } else {
                newState.rules[0].isPassed = false;
                setFullName({ ...newState });
            }

            // checks the word count and make sure the lenght is not less than two
            const wordCount = value.split(' ').filter(word => word !== '').length;
            if (wordCount >= 2 && value.includes(' ')) {
                newState.rules[1].isPassed = true;
                setFullName({ ...newState });
            } else {
                newState.rules[1].isPassed = false;
                setFullName({ ...newState });
            }
        };

        handleFullNameValidationError();
    }, [fullName.value]);

    useEffect(() => {
        // handles validation error for fullname field
        const handlePhoneNumberValidationError = () => {
            // because the phonenumber object is depply nested, we clone the object so that we can change the isPassed trigger
            let newPhoneNumberCopy = JSON.parse(JSON.stringify(phoneNumber));
            /**
             * - the regex matches
             * - the first number should be 0 and a lenght of 1,
             * - second number should match between 7 and 9,
             * - the third number should match either 1 0r 0
             * - the fourth number should be any decimals
             * - and the remaining number should match any number
             */
            const regex = /^([0]{1})([7-9]{1})([0|1]{1})([\d]{1})([\d]{7})$/g;

            // check the length of the value
            if (phoneNumber.value.length >= 11) {
                // allow only 11 charachters
                newPhoneNumberCopy = {
                    ...newPhoneNumberCopy, value: newPhoneNumberCopy.value.slice(0, 11),
                };
                // allow the second rule of allowed characters to pass
                newPhoneNumberCopy.rules[1].isPassed = true;
                setPhoneNumber(newPhoneNumberCopy);
            } else {
                newPhoneNumberCopy.rules[1].isPassed = false;
                setPhoneNumber(newPhoneNumberCopy);
            }

            //    check to see if number is a nigerian number
            if (regex.test(phoneNumber.value) === true) {
                newPhoneNumberCopy.rules[0].isPassed = true;
                setPhoneNumber(newPhoneNumberCopy);
            } else {
                newPhoneNumberCopy.rules[0].isPassed = false;
                setPhoneNumber(newPhoneNumberCopy);
            }
        };
        handlePhoneNumberValidationError();
    }, [phoneNumber.value]);

    useEffect(() => {
        // watch for all the isPassed values in the state and only allow submittion if all ithe rules are passed
        const handleSubmitButtonToggle = () => {
            //add all the state value to an array
            const stateValues = [
                fullName,
                email,
                password,
                confirmPassword,
                phoneNumber,
                cardNumber,
                expiryDate,
                pin,
            ];
            //create anew array to push only rules that have passed validation
            const PassedValues = [];
            stateValues.map(item => {
                // if there are multiple rules in a Component, map through the rule and push only the rule whose rule has been passed
                if (item.rules) {
                    item.rules.map(i => i.isPassed && PassedValues.push(i));
                } else if (item.isPassed) {
                    PassedValues.push(item);
                }
            });
            // check if you have 10 isPassed truthy PassedValues, only then should the button be enabled
            return PassedValues.length === 10 ? setHasAllFieldPassedValidation(true) : setHasAllFieldPassedValidation(false);
        };

        handleSubmitButtonToggle();
    }, [fullName,
        email,
        password,
        confirmPassword,
        phoneNumber,
        cardNumber,
        expiryDate,
        pin]);

    /**
     *This function sets the states of the fullname input by grabing the typed value on the input
     * @param {object} e the events called when a change is occuring n the component
     */
    const handleFullName = e => {
        setFullName({
            ...fullName, value: e.target.value,
        });
    };

    /**
       *This function sets the states of the email input by grabing the typed value on the input
       * @param {object} e the events called when a change is occuring on the component
       */
    const handleEmail = e => {
        // deep clone the object
        const newEmail = JSON.parse(JSON.stringify(email));
        newEmail.value = e.target.value;
        newEmail.isPassed = handleEmailValidity(e);
        setEmail(newEmail);
    };

    /**
    *This function sets the states of the password input by grabing the typed value on the input
    * @param {object} e the events called when a change is occuring on the component
    */
    const handlePassword = e => {
        // deep clone the object
        const newPassword = JSON.parse(JSON.stringify(password));
        newPassword.value = e.target.value;
        newPassword.isPassed = handlePasswordValidity(e);
        setPassword(newPassword);
    };

    /**
    *This function sets the states of the confirm password input by grabing the typed value on the input and comparing against the password field
    * @param {object} e the events called when a change is occuring on the component
    */
    const handleConfirmPassword = e => {
        const handleConfirmPasswordValidity = () => password.value.length > 0 && e.target.value === password.value;
        // deep clone the object
        const newConfirmPassword = JSON.parse(JSON.stringify(confirmPassword));
        newConfirmPassword.value = e.target.value;
        newConfirmPassword.isPassed = handleConfirmPasswordValidity();
        setConfirmPassword(newConfirmPassword);
    };

    /**
    *This function sets the states of the phone number input by grabing the typed value on the input
    * @param {object} e the events called when a change is occuring on the component
    */

    const handlePhoneNumber = e => {
        isAValidNumber(e.target.value) && setPhoneNumber(
            { ...phoneNumber, value: e.target.value }
        );
    };


    /**
    *This function sets the states of the reditcard input by grabing the typed value on the input
    * @param {object} e the events called when a change is occuring on the component
    */
    const handleCardNumberChange = e => {
        const cardValue = e.target.value;
        // check the values, match group of 4 values and push to an array
        const cardValueArray = cardValue.match(/(\d{1,4})/g) || [];
        //    join the array with a space
        const formattedCardValue = cardValueArray.join(' ');

        // check for values length and set value accordingly
        if (e.target.value.length <= 19) {
            setCardNumber({ ...cardNumber, value: formattedCardValue, isPassed: false });
        } if (e.target.value.length === 19) {
            setCardNumber({ ...cardNumber, isPassed: true, value: formattedCardValue });
        }
    };

    /**
    *This function sets the states of the expiry date  input by grabing the typed value on the input
    * @param {object} e the events called when a change is occuring on the component
    */
    const handleExpiryDateChange = e => {
        const expiryDateValue = e.target.value;
        // check the values, match group of 4 values and push to an array
        const expiryDateValueArray = expiryDateValue.match(/(\d{1,2})/g) || [];

        //    join the array with a '/'
        const formattedExpiryDateValue = expiryDateValueArray.join('/');

        if (e.target.value.length <= 5) {
            setExpiryDate({
                ...expiryDate,
                value: formattedExpiryDateValue,
                isPassed: false,
            });
        } if (e.target.value.length === 5) {
            setExpiryDate({
                ...expiryDate,
                isPassed: true,
                value: formattedExpiryDateValue,
            });
        }
    };


    /**
    *This function sets the states of the pin input by grabing the typed value on the input
    * @param {object} e the events called when a change is occuring on the component
    */
    const handlePinChange = e => {
        //first only allow numbers
        if (isAValidNumber(e.target.value)) {
            // only allow input and value change if the value is less than or equal to 4
            if (e.target.value.length <= 4) {
                setPin({
                    ...pin,
                    value: e.target.value,
                    isPassed: false,
                });
            } if (e.target.value.length === 4) {
                setPin({
                    ...pin,
                    isPassed: true,
                    value: e.target.value,
                });
            }
        }
    };

    /**
    *This function handles the submission of the form
    * @param {object} e the events called when a change is occuring on the component
    */
    const handleSubmit = e => {
        e.preventDefault();
        // if the form has passed validation and the submit button clicked, navigate to the dashboard
        if (hasAllFieldPassedValidation === true) {
            history.push('/dashboard');
        }
    };

    return (
        <StyledMainContainer>
            <StyledForm>
                <FullNameInput
                    showFullNameValidationError={showFullNameValidationError}
                    setShowFullNameValidationError={setShowFullNameValidationError}
                    handleFullName={handleFullName}
                    fullName={fullName}
                />
                <EmailInput
                    showEmailValidationError={showEmailValidationError}
                    email={email}
                    handleEmail={handleEmail}
                    setShowEmailValidationError={setShowEmailValidationError}
                />
                <PhoneNumberInput
                    showPhoneNumberValidationError={showPhoneNumberValidationError}
                    phoneNumber={phoneNumber}
                    handlePhoneNumber={handlePhoneNumber}
                    setShowPhoneNumberValidationError={setShowPhoneNumberValidationError}
                />
                <PasswordInput
                    showPasswordValidationError={showPasswordValidationError}
                    password={password}
                    handlePassword={handlePassword}
                    setShowPasswordValidationError={setShowPasswordValidationError}
                />

                <ConfirmPasswordInput
                    showConfirmPasswordValidationError={showConfirmPasswordValidationError}
                    confirmPassword={confirmPassword}
                    handleConfirmPassword={handleConfirmPassword}
                    setShowConfirmPasswordValidationError={setShowConfirmPasswordValidationError}
                />
                <CardNumberInput
                    showCardNumberValidationError={showCardNumberValidationError}
                    cardNumber={cardNumber}
                    handleCardNumberChange={handleCardNumberChange}
                    setShowCardNumberValidationError={setShowCardNumberValidationError}
                />
                <div className="card_pin_date">
                    <ExpiryDateInput
                        showExpiryValidationError={showExpiryValidationError}
                        expiryDate={expiryDate}
                        handleExpiryDateChange={handleExpiryDateChange}
                        setShowExpiryValidationError={setShowExpiryValidationError}
                    />

                    <PinInput
                        showPinValidationError={showPinValidationError}
                        pin={pin}
                        handlePinChange={handlePinChange}
                        setShowPinValidationError={setShowPinValidationError}
                    />
                </div>
                <StyledButton onClick={handleSubmit} disabled={!hasAllFieldPassedValidation}>
                    submit
                </StyledButton>
            </StyledForm>
        </StyledMainContainer>
    );
};
