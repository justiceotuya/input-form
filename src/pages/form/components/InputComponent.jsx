import React, { useState, useEffect } from 'react';
import { StyledInputRule, StyledInputContainer } from '../style';

export const InputComponent = ({
    ValidationError,
    inputValue,
    handleValueChange,
    setShowValidationError,
    type,
    name,
    placeholder,
    label,
}) => {
    const {
        rule, isPassed, value, rules,
    } = inputValue;
    return (
        <StyledInputContainer
            ValidationError={ValidationError}
            isPassed={rules ? null : isPassed}
        >
            <input
                type={type}
                name={name}
                required
                id={name}
                value={value}
                onChange={handleValueChange}
                placeholder={placeholder}
                onBlur={() => setShowValidationError(false)}
                onFocus={() => setShowValidationError(true)}
            />
            <label htmlFor={name}>{label}</label>
            {
                rules ? (
                    <div
                        className="rule"
                    >
                        {
                            rules.map(item => {
                                const { id, rule, isPassed } = item;
                                return (
                                    <StyledInputRule
                                        key={id}
                                        className={`rule-${id}`}
                                        isPassed={isPassed}
                                    >
                                        {rule}
                                    </StyledInputRule>
                                );
                            })
                        }
                    </div>
                ) : <p>{rule}</p>
            }
        </StyledInputContainer>

    );
};
