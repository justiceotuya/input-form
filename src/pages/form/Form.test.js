import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import renderer from 'react-test-renderer';
import { useHistory } from 'react-router-dom'

import { Form } from './Form';
import InputComponent from '../../components/inputComponent'
jest.mock('react-router-dom', () => ({
	useHistory: () => ({
		push: jest.fn(),
	}),
}));

let props = {
	label: 'test',
	name: 'test',
	placeholder: 'test',
	type: 'text',
	inputValue: {
		isPassed: false,
		rules: [{
			id: 1,
			isPassed: true,
			value: ''
		}],
		rule: 'test'
	},
	setShowValidationError: (p => 'p'),
	handleValueChange: (p => 'p'),
	ValidationError: false,
	value: ''
}
const setup = () => {
	const utils = render(<InputComponent  {...props} />)
	const input = utils.getByLabelText('test')
	return {
		input,
		...utils,
	}
}


it('render correctly Form', () => {
	const history = useHistory()
	const tree = renderer.create(<Form />).toJSON();
	expect(tree).toMatchSnapshot();
});

it('render correctly InputComponent', () => {
	const tree = renderer.create(<InputComponent {...props} />).toJSON();
	expect(tree).toMatchSnapshot();
});
