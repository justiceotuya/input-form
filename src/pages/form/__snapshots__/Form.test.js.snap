// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly Form 1`] = `
<main
  className="style__StyledMainContainer-eru9ex-0 caGFYz"
>
  <form
    className="style__StyledForm-eru9ex-1 bsFvrx"
  >
    <div
      className="style__StyledInputContainer-sc-1k0y8dv-0 gfDwB fullName fullName"
    >
      <label
        htmlFor="fullName"
      >
        Full Name
      </label>
      <input
        id="fullName"
        name="fullName"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Adamu Chiroma Adekunle"
        required={true}
        type="text"
        value=""
      />
      <div
        className="rule"
      >
        <p
          className="style__StyledInputRule-sc-1k0y8dv-1 hgRmGt rule-1"
        >
          Name must not be less that 2 characters
        </p>
        <p
          className="style__StyledInputRule-sc-1k0y8dv-1 hgRmGt rule-2"
        >
           Full Name must include a space and then second name
        </p>
      </div>
    </div>
    <div
      className="style__StyledInputContainer-sc-1k0y8dv-0 gfDwB email email"
    >
      <label
        htmlFor="email"
      >
        Email
      </label>
      <input
        id="email"
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="email@address.com"
        required={true}
        type="email"
        value=""
      />
      <p>
        this must an email
      </p>
    </div>
    <div
      className="style__StyledInputContainer-sc-1k0y8dv-0 gfDwB phoneNumber phoneNumber"
    >
      <label
        htmlFor="phoneNumber"
      >
        Phone number
      </label>
      <input
        id="phoneNumber"
        name="phoneNumber"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="08011111111"
        required={true}
        type="text"
        value=""
      />
      <div
        className="rule"
      >
        <p
          className="style__StyledInputRule-sc-1k0y8dv-1 hgRmGt rule-1"
        >
          Must be a Nigerian Phone number
        </p>
        <p
          className="style__StyledInputRule-sc-1k0y8dv-1 hgRmGt rule-2"
        >
          must not be longer or shorter than 11 characters
        </p>
      </div>
    </div>
    <div
      className="style__StyledInputContainer-sc-1k0y8dv-0 gfDwB password password"
    >
      <label
        htmlFor="password"
      >
        Password
      </label>
      <input
        id="password"
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="****************"
        required={true}
        type="password"
        value=""
      />
      <p>
        This password must contain at least one uppercase character,
    one number, special character and not shorter than 6 characters
      </p>
    </div>
    <div
      className="style__StyledInputContainer-sc-1k0y8dv-0 gfDwB confirmPassword confirmPassword"
    >
      <label
        htmlFor="confirmPassword"
      >
        Confirm Password
      </label>
      <input
        id="confirmPassword"
        name="confirmPassword"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="****************"
        required={true}
        type="password"
        value=""
      />
      <p>
        This must be the same as Password
      </p>
    </div>
    <div
      className="style__StyledInputContainer-sc-1k0y8dv-0 gfDwB cardNumber cardNumber"
    >
      <label
        htmlFor="cardNumber"
      >
        Card Number
      </label>
      <input
        id="cardNumber"
        name="cardNumber"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="XXXX XXXX XXXX XXXX"
        required={true}
        type="text"
        value=""
      />
      <p>
        card number must be 16 digits
      </p>
    </div>
    <div
      className="card_pin_date"
    >
      <div
        className="style__StyledInputContainer-sc-1k0y8dv-0 gfDwB expiryDate expiryDate"
      >
        <label
          htmlFor="expiryDate"
        >
          Expiry Date
        </label>
        <input
          id="expiryDate"
          name="expiryDate"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="MM/YY"
          required={true}
          type="text"
          value=""
        />
        <p>
          card number must be 4 digits
        </p>
      </div>
      <div
        className="style__StyledInputContainer-sc-1k0y8dv-0 gfDwB pin pin"
      >
        <label
          htmlFor="pin"
        >
          Pin
        </label>
        <input
          id="pin"
          name="pin"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="****"
          required={true}
          type="password"
          value=""
        />
        <p>
          pin number must be 4 digits
        </p>
      </div>
    </div>
    <button
      className="style__StyledButton-eru9ex-2 jEXebJ"
      disabled={true}
      onClick={[Function]}
    >
      submit
    </button>
  </form>
</main>
`;

exports[`render correctly InputComponent 1`] = `
<div
  className="style__StyledInputContainer-sc-1k0y8dv-0 gfDwB test test"
>
  <label
    htmlFor="test"
  >
    test
  </label>
  <input
    id="test"
    name="test"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="test"
    required={true}
    type="text"
  />
  <div
    className="rule"
  >
    <p
      className="style__StyledInputRule-sc-1k0y8dv-1 cpaqwM rule-1"
    />
  </div>
</div>
`;
